packages/web-frontend/src/components/View/MatrixWrapper/MatrixWrapper.js

```diff
@@ -154,6 +157,12 @@ const buildMatrixDataFromViewContent = viewContent => {
   };
 };
 
+const getPlaceholder = (presentationOptions, categoryPresentationOptions) => {
+  if (!presentationOptions && !categoryPresentationOptions) return textOnlyMatrixPlaceholder;
```

Comment:
Logic from [code](https://github.com/beyondessential/tupaia/blob/954a73e64d3a5df1b0da147d78b109daf0d85d92/packages/web-frontend/src/components/View/MatrixWrapper/components/Matrix.js#L142)

Reply:
Would be better to pull this out into a named boolean, or a named function (like the place you got it from does). That makes the condition a lot easier to read inline, because it just looks like 
```suggestion
  if (!isUsingDots) return textOnlyMatrixPlaceholder;
```

And then the logic for working out `isUsingDots` is clearly defined on another line/in another place.

Ideally, this logical block would be shared across both components. A couple of options there:
- pull out a util function that is common to both
- work it out in the wrapper and pass the result down to the child

Reply:
Thanks Edwin, I agree having a named boolean is mush easier to read, definitely will put it in. 👍 
And I was thinking to use `isUsingDots` here then pass in to children component, but there is different `isUsingDots` logic for `Row` and `RowGroup` component, which in `Row` it is a dynamic variable that can be customed to show text for special index.

Reply:
Cool, in that case perhaps pull out a util function that can be called from both places :-)